sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -743,6 +747,16 @@ private void reportBacklog() {
     backlogElementsOfSplit.set(splitBacklogMessages);
   }
 
+  private void reportBacklogMetrics() {
```

Comment:
Looks like this can be merged with `reportBacklog` (potentially rename that method to `reportBacklogMetrics` `updateBacklogMetrics`).

Reply:
No, I explicitly moved it out to be separate, since reportBacklog() is called twice, and we only need to do this once (when we advance to the next record).